src/common/Makefile.am

```diff
@@ -100,7 +100,10 @@ noinst_LTLIBRARIES += libcommon_internal.la
 libcommon_api_la_SOURCES = \
 	common/buffer.cc
 if LINUX
+if !ENABLE_XIO
+# XioMessenger.cc uses get_xio_mp() in buffer.cc
 libcommon_api_la_CXXFLAGS = -fvisibility=hidden -fvisibility-inlines-hidden
+endif # !ENABLE_XIO
```

Comment:
I think we might want a better way of handling this — IIRC we've hidden those symbols deliberately to prevent users doing things they shouldn't. XioMessenger is definitely part of the system internals; is there some way we can grant it (and only it) access? Or build them together or something?